base_global_discount/models/global_discount.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class GlobalDiscount(models.Model):
+    _name = 'global.discount'
+    _description = 'Global Discount'
+    _order = "sequence, id desc"
+
+    sequence = fields.Integer(
+        help='Gives the order to apply discounts',
+    )
+    name = fields.Char(
+        string='Discount Name',
+        required=True,
+    )
+    color = fields.Integer('Color Index', default=10)
+    discount_type = fields.Selection(
+        selection=[
+            ('amount', 'Fixed amount'),
+            ('percentage', 'Percentage'),
+        ],
+        default='percentage',
+        required='True',
+        string='Discount Type',
+    )
+    discount = fields.Float(
+        digits=dp.get_precision('Product Price'),
+        default=0.0,
+    )
+    discount_scope = fields.Selection(
+        selection=[
+            ('sale', 'Sale'),
+            ('purchase', 'Purchase'),
+        ],
+        required='True',
+        string='Discount Scope',
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
```

Comment:
Add default to current user company

Reply:
Should not they be globally available by default?

Reply:
Not really, and this is not the default behavior in other places. Imagine this scenario: you have a multi-company DB, and you are a user with permissions for only one company and you are not even aware of that field (as it's hidden by default with the multi-company group). You create one record, thinking that is only for your company, and it appears in other companies! So it should be the contrary: belonging to one company by default, and having to be changed to empty for doing it global.

Reply:
ok then